# **Plotly**
Plotly interaktif görseller oluşturmamıza yarayan açık kaynak kodlu d3.js üzerine inşa edilmiş bir kütüphanedir ve Matlab, R ve Python API’leri bulunmaktadır.Böylelikle kolaylıkla interaktif görseller oluşturulabilir.

Plotly'da grafik üzerindeki konuma gelindiğinde ait olunan değeri gösterir.

Plotly kütüphanesi aynı zamanda çeşitli grafikler de sunmaktadır;

*   Kontur grafikleri, log grafikleri vb. gibi Bilimsel Grafikler.
*   Huni çizelgeleri, şamdan çizelgeleri vb. gibi Finansal Grafikler
*   Kabarcık haritası, yoğunluk haritası vb. haritalar.


Plotly kütüphanesinde ;

*   line plot
*   scatter plot
*   area chart
*   bar chart
*   error bar
*   box plot
*   histogram
*   heatmap
*   subplot
*   multiple-axes
*   polar chart
*   bubble chart görselleştirilmeleri yapılır.








Plotly kütüphanesini çalıştırabilmek için bazı kurulumlar yapılması gereklidir.

In [ ]:
!pip install --upgrade plotly 

In [ ]:
!pip install jupyter-dash

graph_objt modülü ile farklı türde grafikler oluşturulur. 

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from plotly.offline import iplot
import plotly.graph_objs as go

from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
timesData = pd.read_csv("/content/drive/MyDrive/data/timesData.csv")

Bu data üniversitelerin dünya sıralamasını içermektedir.

Datada bulunan sütunların içeriği şunlardır;

*   **world_rank** ->Dünya sıralaması
*   **university_name** -> Üniversite ismi
*   **country** -> Üniversitenin hangi ülkede olduğunun bilgisi
*   **teaching** -> Üniversitenin öğretme değeri (öğrenme oranı)
*   **international** -> üniversitenin uluslararası skoru
*   **research** -> üniversitenin araştırma skoru
*   **citations** -> üniversitenin alıntı yapma skoru
*   **income** -> endüstri geliri için üniversite skoru (bilgi transferi)
*   **total_score** -> üniversitenin toplam skoru (derecesini belirlemek için kullanılır.)
*   **num_students** -> üniversitedeki toplam öğrenci sayısı
*   **student_staff_ratio** -> çalışan sayısının öğrenci sayısına bölümü
*   **international_students** -> üniversitedeki Uluslararası öğrencilerin yüzdesi
*   **female_male_ratio** -> kadın öğrencinin erkek öğrenciye oranı
*   **year** -> sıralama yılı
























In [5]:
timesData.head()

world_rank                        university_name                   country  \
0          1                     Harvard University  United States of America   
1          2     California Institute of Technology  United States of America   
2          3  Massachusetts Institute of Technology  United States of America   
3          4                    Stanford University  United States of America   
4          5                   Princeton University  United States of America   

   teaching international  research  citations income total_score  \
0      99.7          72.4      98.7       98.8   34.5        96.1   
1      97.7          54.6      98.0       99.9   83.7        96.0   
2      97.8          82.3      91.4       99.9   87.5        95.6   
3      98.3          29.5      98.1       99.2   64.3        94.3   
4      90.9          70.3      95.4       99.9      -        94.2   

  num_students  student_staff_ratio international_students female_male_ratio  \
0       20,152                  8.9                    25%               NaN   
1        2,243                  6.9                    27%           33 : 67   
2       11,074                  9.0                    33%           37 : 63   
3       15,596                  7.8                    22%           42 : 58   
4        7,929                  8.4                    27%           45 : 55   

   year  
0  2011  
1  2011  
2  2011  
3  2011  
4  2011

In [6]:
timesData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2603 entries, 0 to 2602
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   world_rank              2603 non-null   object 
 1   university_name         2603 non-null   object 
 2   country                 2603 non-null   object 
 3   teaching                2603 non-null   float64
 4   international           2603 non-null   object 
 5   research                2603 non-null   float64
 6   citations               2603 non-null   float64
 7   income                  2603 non-null   object 
 8   total_score             2603 non-null   object 
 9   num_students            2544 non-null   object 
 10  student_staff_ratio     2544 non-null   float64
 11  international_students  2536 non-null   object 
 12  female_male_ratio       2370 non-null   object 
 13  year                    2603 non-null   int64  
dtypes: float64(4), int64(1), object(9)
memor

times datası; 
*   Dataframe türündedir.
*   2603 sample bulunmaktadır.
*   14 sütundan oluşmakta, bu sütunların 4 tanesi float türünde, 1 tanesi integer türünde, 9 tanesi string türünde veri barındırmaktadır.
*   num_students, student_staff_ratio, international_students ve female_male_ratio sütunlarında missing data olduğu görülmektedir.



